### ISSUE TITLE
Why is L_upper in Eq. (9) implemented as a cross-entropy function?

### ISSUE THREAD

**Issue by hongliang-wei** [bbbdylan/proda#1]
Thank you for your great work.Not sure if I understand correctly , why is L_upper in Eq. (9) implemented as a cross-entropy function?

**Comment by lynlynlyn** [comment]
L_upper (Eq. 6) can be considered as adjusting the output logits (by adding a covariance term). Note that the covariance term of the positive term is zero.

**Comment by hongliang-wei** [comment]
I am still a little confused, can you tell me where is the code to calculate the second term of the denominator in Eq. (6)( (z_i)^T*A*z_i )? Thank you!

**Comment by hongliang-wei** [comment]
I understand, thank you!
